### PR TITLE
Add test for tactic in rule filename

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
related to #380

## Summary
Rule filenames with the tactic in them were inconsistent. The general rule is that the first element in the `threat` array is considered the most representative tactic and so also the one used in the filename.

This test validates that all (non-ml) rules with a threat mapping have the proper tactic in the filename.

cc: @seth-goodwin

### CODE REVIEW

File: tests/test_all_rules.py [hunk 1]
```diff
@@ -268,3 +269,27 @@ def test_required_tags(self):
...
+        for rule in rules:
+            rule_path = Path(rule.path).resolve()
+            filename = rule_path.name
+
+            if rule_path.parent.name == 'ml':
+                continue
+
+            threat = rule.contents.get('threat')
+
+            if threat:
```
Comment (rw-access) [inline tests/test_all_rules.py:290]: we should avoid breaking changes for any forks of the repo with custom rules. here's one way:
```suggestion
            if threat and rule.metadata.get("author") == "Elastic":
```

**Review by bm11100 — APPROVED** [review]
👍 cool wit it

File: tests/test_all_rules.py (hunk 1 above)
Comment (brokensound77) [inline tests/test_all_rules.py:290]:
```suggestion
            threat = rule.contents.get('threat')
            author = rule.contents.get('author')

            if threat and author and 'Elastic' in author:
```

Work for you @rw-access?

Comment (rw-access) [reply tests/test_all_rules.py:290]: good catch. forgot about the array.
 minor tweak:
```suggestion
            threat = rule.contents.get('threat', [])
            authors = rule.contents.get('author', [])
            
            if threat and 'Elastic' in authors:
```

Comment (brokensound77) [reply tests/test_all_rules.py:290]: wfm 👍

**Review by rw-access — APPROVED** [review]

**Pull request merged**